wordpress-mobile/WordPress-Android
# MSD - Today's Stats Card - Metrics

Fixes #15751

References: p1643829887600439-slack-C0290FLA0RM

This PR ensures that metrics are properly added for `Today's Stats Dashboard Card`. 

To Test:

Setup

- Launch the app
- Navigate to Me -> App Settings -> Debug Settings
- Ensure that the `MySiteDashboardTodaysStatsCardFeatureConfig` is to set to on
- Restart the app if needed
- Navigate to Me -> App Settings -> Privacy Settings
- Toggle the "collect information" switch to on
- Navigate to the My Site tab

Then follow instructions on how to enable
- showing today's stats card with data in #15834,  
- showing today's stats card with error in #15893,  and
- clicking today's stats card actions in #15908

Notice the following trackings in logs:

Event| Tracking
------|-----------
MY_SITE_DASHBOARD_CARD_SHOWN (without error) | `🔵 Tracked: my_site_dashboard_card_shown, Properties: {"type”:”stats”,”subtype":"todays_stats"}`
MY_SITE_DASHBOARD_CARD_FOOTER_ACTION_TAPPED | `🔵 Tracked: my_site_dashboard_card_footer_action_tapped, Properties: {"type":"stats","subtype":"todays_stats"}`
MY_SITE_DASHBOARD_CARD_ITEM_TAPPED | `🔵 Tracked: my_site_dashboard_card_item_tapped, Properties: {"type":"stats","subtype":"todays_stats"}`
MY_SITE_DASHBOARD_CARD_SHOWN (with error) | `🔵 Tracked: my_site_dashboard_card_shown, Properties: {"type":"error","subtype":"todays_stats"}`
MY_SITE_MENU_ITEM_TAPPED | `🔵 Tracked: my_site_menu_item_tapped, Properties: {"type":"stats"}`

Additionally, for AB Testing (see p1645013610195379-slack-C0290FLA0RM), notice that today's stats enabled state is tracked as part of the following events:

1. Stats screen accessed

    `🔵 Tracked: stats_accessed, Properties: {"org.wordpress.android.util.config.MySiteDashboardTodaysStatsCardFeatureConfig":true,"blog_id":<>,"is_jetpack":<>,"site_type":"blog"}`

2. My site tab accessed

    `🔵 Tracked: my_site_tab_accessed, Properties: {"org.wordpress.android.util.config.MySiteDashboardTodaysStatsCardFeatureConfig":true,"blog_id":<>,"is_jetpack":<>,"site_type":"blog"}`

Note: Today's stats feature config name is based on [this logic](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/config/FeatureConfig.kt#L20). Since `remoteConfig` is not yet enabled, you'll notice full java class name for the config flag.

**Review Instructions:**

Review by only one reviewer is sufficient.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
